fl4health/clients/ditto_client.py

```diff
@@ -9,31 +9,35 @@
 from torch.optim import Optimizer
 
 from fl4health.checkpointing.client_module import ClientCheckpointModule
-from fl4health.clients.basic_client import BasicClient
-from fl4health.losses.weight_drift_loss import WeightDriftLoss
+from fl4health.clients.adaptive_drift_constraint_client import AdaptiveDriftConstraintClient
 from fl4health.parameter_exchange.full_exchanger import FullParameterExchanger
+from fl4health.reporting.metrics import MetricsReporter
 from fl4health.utils.config import narrow_config_type
 from fl4health.utils.losses import EvaluationLosses, LossMeterType, TrainingLosses
 from fl4health.utils.metrics import Metric
 from fl4health.utils.typing import TorchFeatureType, TorchInputType, TorchPredType, TorchTargetType
 
 
-class DittoClient(BasicClient):
+class DittoClient(AdaptiveDriftConstraintClient):
     def __init__(
         self,
         data_path: Path,
         metrics: Sequence[Metric],
         device: torch.device,
         loss_meter_type: LossMeterType = LossMeterType.AVERAGE,
         checkpointer: Optional[ClientCheckpointModule] = None,
-        lam: float = 1.0,
+        metrics_reporter: Optional[MetricsReporter] = None,
+        progress_bar: bool = False,
     ) -> None:
         """
         This client implements the Ditto algorithm from Ditto: Fair and Robust Federated Learning Through
         Personalization. The idea is that we want to train personalized versions of the global model for each client.
         So we simultaneously train a global model that is aggregated on the server-side and use those weights to also
         constrain the training of a local model. The constraint for this local model is identical to the FedProx loss.
 
+        NOTE: lambda, the drift loss weight, is initially set and potentially adapted by the server akin to the
```

Comment:
Maybe it worth mentioning that using adaptation is optional and specified in the strategy. Just so users are aware they can use the original Ditto if they need to do so

Reply:
Sure!